node_modules/oae-authz/lib/api.js

```diff
@@ -711,48 +872,92 @@ var _getIndirectPrincipalMembershipsByExplosion = function(principalId, start, l
 };
 
 /**
- * Given some principal, get their entire group membership ancestry. In other words, get all the groups to which they belong,
- * and the groups to all those groups belong, and so on.
+ * Create a group memberships graph for all the provided principals. The resulting graph is
+ * guaranteed to contain all principal ids, even if some or all of them don't have any memberships.
+ * The result is a graph that is possibly several disjoint graphs, as it's not necessary that all
+ * provided principals have a memberships graph that intersect with one another
```

Comment:
Can you add that this will return the full (so both direct and indirect) memberships?